Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -1704,6 +1704,37 @@ class ByteBufferTest: XCTestCase {
         XCTAssertEqual(0xcc, slice.getInteger(at: 0, as: UInt8.self))
         XCTAssertEqual(0xdd, slice.getInteger(at: slice.writerIndex - 1, as: UInt8.self))
     }
+    
+    func testSliceOnSliceAfterHitting16MBMark() {
+        // This test ensures that a slice will get a new backing storage if its start is more than
+        // 16MiB after the originating backing storage.
+        
+        // create a buffer with (16MiB - 1byte) + 2 bytes = 16MiB + 1 byte
+        let inputBufferLength = Int(_UInt24.max) + 2
```

Comment:
Just hardcode the value here. The `_` means that `_UInt24` is meant to be private. So `let inputBufferLength =  1 << 24 + 1` or similar.